src/engine/IndexScan.h

```diff
@@ -231,6 +238,34 @@ class IndexScan final : public Operation {
       std::optional<std::vector<CompressedBlockMetadata>> blocks =
           std::nullopt) const;
   std::optional<Permutation::MetadataAndBlocks> getMetadataForScan() const;
+
+  // If the `varsToKeep_` member is set, meaning that this `IndexScan` only
+  // returns a subset of this actual columns, return the subset of columns that
+  // has to be applied to the "full" result (without any columns stripped) to
+  // get the final result. Throws if `varsToKee_` is `nullopt`.
+  std::vector<ColumnIndex> getSubsetForStrippedColumns() const;
+
+  // Return a lambda that takes an `idTable` that has the result without any
+  // columns stripped, and applies the column subset that leads to the correct
+  // stripping of the columns. This function can also be used if no columns are
+  // stripped and hence `varsToKeep_` is `nullopt`.
+  auto makeApplyColumnSubset() const {
```

Comment:
Please add a comment explaining that ideally we wouldn't write the columns in the first place so this becomes redundant.